### PR TITLE
FR-78 get list of ideas func

### PR DESCRIPTION
**Pull request by VladimirKarin**
Created function, that get a lists of ideas. 
Tested in terminalApp.

### CODE REVIEW

File: src/data/ideas.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// ideas.js
+const axios = require('axios');
```
Comment (fedosejev) [inline src/data/ideas.js]: Can you implement data fetching without `axios`, but with `fetch()` API?

See the documentation: https://nextjs.org/docs/app/building-your-application/data-fetching

File: tests/terminalApp.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    .then((response) => {
+        // The server responded with a successful status code (2xx)
+        const ideas = response.data;
+        console.log('Ideas received:', ideas);
+        // Now you can do something with the ideas data on the client-side
+    })
+    .catch((error) => {
+        // Handle errors, e.g., server is down, network issues, etc.
+        console.error('Error fetching ideas:', error);
+    });
```
Comment (fedosejev) [inline tests/terminalApp.js]: The purpose of this Terminal app is to test your `getIdeas` function, but you're not doing it here

File: src/data/ideas.js (hunk 1 above)
Comment (VladimirKarin) [reply src/data/ideas.js]: Yeah, I've read that. But thought I could write it without await and async. Fixed.

File: tests/terminalApp.js (hunk 2 above)
Comment (VladimirKarin) [reply tests/terminalApp.js]: To be honest this was the hardest part. I've had a chat with very wise sinsAI, he suggested me this kind of way. It worked.

File: tests/terminalApp.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const getIdeas = require('../src/data/ideas');
+const URL = 'https://fund-raising-project-server-side.azurewebsites.net/ideas';
```
Comment (fedosejev) [inline tests/terminalApp.js]: What do you need this URL for?

File: package.json [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+    "version": "0.1.0",
+    "private": true,
+    "scripts": {
+        "dev": "next dev",
+        "build": "next build",
+        "start": "next start",
+        "lint": "next lint"
+    },
+    "dependencies": {
+        "axios": "^1.4.0",
```
Comment (fedosejev) [inline package.json]: Don't forget to remove unused dependencies

File: src/data/ideas.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const ideasUrl =
+    'https://fund-raising-project-server-side.azurewebsites.net/ideas';
+
+async function getIdeas() {
+    try {
+        const response = await fetch(ideasUrl);
+        if (!response.ok) {
+            throw new Error('Failed to fetch data');
+        }
+        return await response.json();
```
Comment (fedosejev) [inline src/data/ideas.js]: Why do you need `await` here?

File: package.json [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
+        "eslint": "8.45.0",
+        "eslint-config-next": "13.4.12",
+        "next": "13.4.12",
+        "node-fetch": "^3.3.2",
```
Comment (fedosejev) [inline package.json]: What do you need this package for? I don't see it being used anywhere

File: tests/terminalApp.js (hunk 3 above)
Comment (VladimirKarin) [reply tests/terminalApp.js]: That one left after previouse version. Fixed.

File: package.json (hunk 4 above)
Comment (VladimirKarin) [reply package.json]: Fixed.

File: src/data/ideas.js (hunk 5 above)
Comment (VladimirKarin) [reply src/data/ideas.js]: Async\await is a bit complicated for me, I'll havev to read some more about it. Fixed.

File: package.json (hunk 6 above)
Comment (VladimirKarin) [reply package.json]: node-fetch? Yeah, I've tried to install it but, after a while found out, that it is built in in Node.js 18. Fixed

Comment (fedosejev) [reply package.json]: The `node-fetch` dependency is still there

Comment (VladimirKarin) [reply package.json]: fixed

**Review by fedosejev — APPROVED** [review]

**Pull request merged**